core/src/lib/internal/CoreStitchAppClient.ts

```diff
@@ -16,10 +17,9 @@ export default class CoreStitchAppClient {
   }
 
   public callFunctionInternal<T>(name: string, ...args: any[]): Promise<T> {
-    // TODO: STITCH-1510
-    return this.authRequestClient.doAuthenticatedJSONRequest(
+    return EJSON.parse(this.authRequestClient.doAuthenticatedJSONRequest(
```

Comment:
I don't think this will work. `doAuthenticatedJSONRequest()` returns a `Promise<Response>` so I think you'll need to get the result of the promise and call EJSON.parse on the response body.

Reply:
Also I think this logic also needs to go in `CoreStitchService`

Reply:
I agree about point 1, but point 2 I'm not sure about. Seems like not everything from `CoreStitchService` may be extJson.

Reply:
just `doAuthenticatedJSONRequest`? Are there any endpoints that don't accept a JSON body, but still return extended JSON?

Reply:
There are not in the client API. All functions require a JSON body.